staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -296,12 +296,21 @@ func canIdentifyUser(config restclient.Config) bool {
 
 // Namespace implements ClientConfig
 func (config *DirectClientConfig) Namespace() (string, bool, error) {
-	if config.overrides != nil && config.overrides.Context.Namespace != "" {
+	// if an override is provided and it starts with namespaces/ (as generated by `kubectl get ns -o name`), strip it off
+	// (this is intentionally targeted towards interactive use only)
+	namespace := config.overrides.Context.Namespace
```

Comment:
should check `if config.overrides != nil` before this line, or you may get a panic.